kube-runtime/src/controller/mod.rs

```diff
@@ -593,6 +593,55 @@ where
     /// The [`ListParams`] refer to the possible subset of `Watched` objects that you want the [`Api`]
     /// to watch - in the Api's configured scope - and run through the custom mapper.
     /// To watch the full set of `Watched` objects in given the `Api` scope, you can use [`ListParams::default`].
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use kube::runtime::{Controller, controller::Action, reflector::ObjectRef};
+    /// # use kube::api::{Api, ListParams};
+    /// # use kube::ResourceExt;
+    /// # use k8s_openapi::api::core::v1::{ConfigMap, Namespace};
+    /// # use futures::StreamExt;
+    /// # use std::sync::Arc;
+    /// # type ApiResource = Namespace;
+    /// # struct Context;
+    /// # async fn reconcile(_: Arc<ConfigMap>, _: Arc<Context>) -> Result<Action, kube::Error> {
+    /// #     Ok(Action::await_change())
+    /// # };
+    /// # fn error_policy(_: Arc<ConfigMap>, _: &kube::Error, _: Arc<Context>) -> Action {
+    /// #     Action::await_change()
+    /// # }
+    /// # async fn doc(client: kube::Client) -> Result<(), Box<dyn std::error::Error>> {
+    /// # let memcached = Api::<ConfigMap>::all(client.clone());
+    /// # let context = Arc::new(Context);
+    /// Controller::new(memcached, ListParams::default())
+    ///     .watches(
+    ///         Api::<ApiResource>::all(client.clone()),
+    ///         ListParams::default(),
+    ///         |ar| {
```

Comment:
i think we should extract this to an actual fn since it's the docs for how to write that function

Reply:
maybe something like:
`
fn operator_sdk_resource_mapper(ns: &str, api: &Api<Namespace>) -> Option<ObjectRef<ConfigMap>>
```

Reply:
we could even add a link comment inside that function body to link to the operator-sdk page for this particular mapping

Reply:
I think if we have a function that extracts a 'generic' ref, the types should also be generic, which makes the example a lot more complex and imo less readable.

Reply:
yeah, that is fair. particularly if the namespace type was intended to be hidden.